server/auth_function/lambda_function.py

```diff
@@ -207,6 +207,45 @@ def handle_event(db_connection, event) -> event_type.Event:
     for record in cursor:
         role_list.append(record[0])
 
+    # check if login through FAM
```

Comment:
I wonder if we want to check if the user login through FAM itself, which way is better?

- use the cognito_client_id and query in the database to see if the application_name is "FAM"
- hardcode the cognito_client_id for FAM (dev, test, prod), and just check if the given cognito_client_id is FAM's cognito_client_id

We check if user login through FAM is because we only need to query the fam_application_admin table when user login through FAM itself.

Reply:
I feel fine from your implementation. 
However, later for delegated admin, I guess `auth_function` at this condition (when login to FAM) needs to be adjusted further and query new table again for adding extra roles into JWT and we might need to refactor it into better block of code/function?